format/index.md

```diff
@@ -785,6 +794,15 @@ include additional [fields] in the response.
 GET /articles?include=author&fields[articles]=title,body&fields[people]=name
 ```
 
+If both query parameters `include` and `fields[TYPE]` are provided by the client, 
+values of `TYPE` **MUST** either refer to the primary data or be listed in 
+`include`, otherwise the server **MUST** return `400 Bad Request`.
+
+If both query parameters `include` and `fields[TYPE]` are provided by the client,
+and a given related object is requested in `include` but reference to it is not 
+requested in the corresponding `fields[TYPE]`, the server **MUST** return 
+`400 Bad Request`.
```

Comment:
Again, I don't think this is appropriate for heterogeneous collections and should probably be removed.